src/chronify/time_utils.py

```diff
@@ -0,0 +1,86 @@
+"""Functions related to time"""
+
+import logging
+import numpy as np
+from typing import Any
+
+import pandas as pd
+
+from chronify.time import (
+    TimeIntervalType,
+)
+from chronify.exceptions import InvalidParameter
+
+logger = logging.getLogger(__name__)
+
+
+def shift_time_interval(
+    df: "pd.Series[pd.Timestamp]",
```

Comment:
This looks like a very annoying problem. I probably would have just annotated it with `pd.Series` and silenced mypy with `# type: ignore`.

Secondly, is it common among pandas users to use the variable name `df` for Series objects? I find that a little confusing.

Lastly, since this function is essentially numpy and not pandas, would it be better to pass and return numpy NDArray?

Reply:
Took me a while to find this solution. 
Using dfs for series.
The output is not NDArray but is still pd.Series (except for the the old roll_timestamps).